airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -163,8 +175,14 @@ class KubernetesPodOperator(BaseOperator):
     :param termination_grace_period: Termination grace period if task killed in UI,
         defaults to kubernetes default
     :type termination_grace_period: int
+
+    TODO: ``is_delete_operator_pod`` default should be True
```

Comment:
Should we remove this?

If not, can we add more context on why we want it to be `True`?

Reply:
this was a discussion point in the early part of the process for this operator.  @jedcunningham suggested taking the opportunity to fix a few things as part of the refactor.  along with removing some of the duplication, he suggested flipping delete pod to True, and adding support for usage of k8s hook.  if you're cool with both of these changes, i can add them both to this PR.  or we can defer them -- either way i can remove that TODO but let's resolve what to do here first.

Reply:
Almost everyone wants the pods to be cleaned up after it finishes, except when debugging. I'm also just tired of having to always set it, and I can't be the only one that it bothers.

Reply:
Let's change that in a separate PR -- so that this PR just simplifies/refactors things but keeps the same behavior